call-home/src/bin/callhome/collector/report_models.rs

```diff
@@ -363,3 +363,8 @@ impl<'a, T: TryFrom<Record<'a>>> From<&'a EventsRecord> for Option<T> {
         src.record.iter().find_map(|s| T::try_from(Record(s)).ok())
     }
 }
+
+// Define the `is_zero` function to determine if the field should be serialized.
+fn is_zero<T: PartialEq + Default>(value: &T) -> bool {
+    *value == T::default()
```

Comment:
Though suggest you change impl of this function as it's still not doing what's it's suggesting

Reply:
Something like this is clearer..

```rust
fn is_zero(value &i32) -> bool {
  value == &0
}
```

Reply:
done